PaymentButtons/src/main/java/com/paypal/android/paymentbuttons/PayLaterButton.kt

```diff
@@ -49,5 +49,10 @@ class PayLaterButton @JvmOverloads constructor(
 
     init {
         updateLabel(PayPalButtonLabel.PAY_LATER)
```

Comment:
As PayLaterButton extends PayPalButton, we'll have a duplicate initialized event logged.

Reply:
🤔 is there a way to avoid this? I think we want to avoid sending duplicate events.

Reply:
If all of the button classes extend `PayPalButton` could we instead have this initialization event just sent there and use the `fundingType` property passed into the `buttonType`?

Reply:
> If all of the button classes extend `PayPalButton` could we instead have this initialization event just sent there and use the `fundingType` property passed into the `buttonType`?

I had initially put the analytics block in the `PaymentButton` - which is the super class of all these views. The issue I ran into and learned about was the execution order in which Kotlin executes various `constructors`, `init` blocks and the class level variables.
See: [An in-depth look at Kotlin’s initializers](https://medium.com/keepsafe-engineering/an-in-depth-look-at-kotlins-initializers-a0420fcbf546#:~:text=Execution%20order&text=First%2C%20default%20constructor%20arguments%20are,%2C%20top%2Dto%2Dbottom)

I've spent some time on it and wasn't able to figure out a way to make it possible.
I wouldn't mind pairing on this to figure out if it's possible.

Reply:
🤔 that's super interesting and thanks for doing that research/laying it out in a way that's easy to understand! Is there anything specific that we have to have from extending `PayPalPalButton` or can we duplicate some code/extend `PaymentButton` (similar to what it looks like the `PayPalCreditButton` does). Duplicating events for 1 payment button will pretty greatly skew our data. What do you think?

Reply:
I've always felt like we should have one button class. Karthik felt the same way when he was here. I wonder if the button analytics work is starting to reveal tech debt?

Reply:
> 🤔 that's super interesting and thanks for doing that research/laying it out in a way that's easy to understand! Is there anything specific that we have to have from extending `PayPalPalButton` or can we duplicate some code/extend `PaymentButton` (similar to what it looks like the `PayPalCreditButton` does). Duplicating events for 1 payment button will pretty greatly skew our data. What do you think?

Duplicating logic should be doable.

>I've always felt like we should have one button class. Karthik felt the same way when he was here. I wonder if the button analytics work is starting to reveal tech debt?

@sshropshire there was a ticket to convert everything to a single button class.
1. it is a breaking change.
2. we'll likely run into issues with Android framework liking its constructors to not have any extra arguments. I've tried taking a stab at it, but I didn't like the outcome.

Reply:
> Duplicating logic should be doable.

This to me makes sense to tackle in the scope of this PR since currently we'd be sending duplicate events. What do you think?

Reply:
> > Duplicating logic should be doable.
> 
> This to me makes sense to tackle in the scope of this PR since currently we'd be sending duplicate events. What do you think?

made the change :) Please take another look.